### ISSUE TITLE
10.15.4 battery / sleep / trackpad not working 电池状态、休眠、触摸板都不正常

### ISSUE THREAD

**Issue by TooBug** [Azkali/GPD-P2-MAX-Hackintosh#26]
我使用的是@andot 的fork版本，安装系统正常。但是有几个硬件不正常。

I am using @andot 's fork. MacOS was installed successfully. But a few hardware not working.

### 电池状态 Battery Status

无法获取电池状态，在偏好设置中，看到的电量是0%，无法勾选在状态栏显示电池图标，只要一勾上，状态栏中的图标会闪一下立即消失。

Can not get battery status. It is always 0% in system preferences, and can not enable the "show battery icon in status menu bar", once checked, the icon flashed and disappeared very soon.

### 休眠 Sleep

无法合盖休眠，合盖后屏幕还是亮的。

Can not get slept when close the laptop, the screen keeps lighting.

### 触摸板

触摸板在系统中识别不到，偏好设置中看到的是鼠标，鼠标的设置项对触摸板有效。尝试过在PE系统下写入仓库中的驱动，写入完之后没有任何变化。

Can not see trackpad in system preferences, instead mouse preferences takes effect to touchpad. I tried to burn the driver in this repo in a WinPE system, but nothing changed after the new firmware burnt.

尝试过将DSDT_2.aml换成DSDT.aml，换完后系统无法启动，只能换回来。

Tried to replace DSDT.aml with DSDT_2.aml, than it failed to boot, has to change that file back.

刚接触黑苹果，求助，这么多硬件不正常工作，有可能是哪里出现问题……

I am new to hackintosh, please give me some tips, so many hardware not working as expected, where could the problem comes from?

**Comment by TooBug** [comment]
又重启了一次，发现触控屏突然正常了……

**Comment by andot** [comment]
电池状态：参考这篇 https://github.com/Azkali/GPD-P2-MAX-Hackintosh/issues/16 ，把 CFG Lock 禁用。另外，我的 BIOS 是 0.25 的，我不清楚你的 BIOS 版本，也许别的版本有差异。

休眠：可能跟外接的 USB 设备有关，参考这篇：https://github.com/Azkali/GPD-P2-MAX-Hackintosh/issues/5

触摸板：这个设备的触摸板是跟键盘一体的，所以在系统里找不到它的设置，但是它是可以正常工作的。

**Comment by TooBug** [comment]
电池的问题：BIOS是0.29版本。参考了你发的链接，应该是patch成功了，也修改了clover的配置，只开启了Apple RTC，但还是没有任何变化。

**Comment by TooBug** [comment]
可否请教，patch成功后，如何查看CFG Lock的值？我确认一下是否是关闭了。

**Comment by andot** [comment]
你的 BIOS 版本号太高了，两个 DSDT.aml 都没有针对这个版本 BIOS 做过适配。

**Comment by TooBug** [comment]
明白了，谢谢！

那我能做点什么让它尽可能正常工作？重新提取DSDT然后再把补丁打上去有可行性吗？能否提供一些比较有价值的参考资料，我自己试着弄一弄看看？

**Comment by andot** [comment]
https://listenerri.com/2019/10/11/%E9%80%9A%E8%BF%87%E6%89%93DSDT%E8%A1%A5%E4%B8%81%E8%AE%A9%E9%BB%91%E8%8B%B9%E6%9E%9C%E6%98%BE%E7%A4%BA%E7%94%B5%E6%B1%A0%E7%8A%B6%E6%80%81%E4%BF%A1%E6%81%AF/

**Comment by Azkali** [comment]
@andot I invited you as contributor to the repository so you can manageit if you prefer ...
Sorry for the lack of answer lately on the repository, I'm working on other subjects for the moment, but I haven't forgot about this project.

**Comment by andot** [comment]
@Azkali Thank, I have accepted the invitation. 😄

**Comment by miiichiii** [comment]
I had the same situation(no battery status, touch screen, touchpad,)
However, after I removed DSDT.aml, and renamed DSDT_2.aml as DSDT.aml in /EFI/CLOVER/ACPI/patched/, the problem was resolved.

**Comment by Andiweli** [comment]
Sleep is not working for me too :(

**Comment by jschaeke** [comment]
+1, 10.15.5 battery not working too (bios 0.29)

**Comment by Andiweli** [comment]
You have to use DSDT_2.aml and the SSDT-PLNF (instead of the SSDT-PLNF_027).
I am on BIOS 0.29 too and this works for me.

Also keep an eye on updates for other kexts.
Azkali has updated the files today... perhaps you should try this.

**Comment by jschaeke** [comment]
> You have to use DSDT_2.aml and the SSDT-PLNF (instead of the SSDT-PLNF_027).
> Azkali has updated the files today... perhaps you should try this.

Yep I am on latest and tried both DSDT_2 as DSDT.aml. I am wondering how can make those work, I assume it's going through this process: https://www.tonymacx86.com/threads/guide-patching-laptop-dsdt-ssdts.152573/
The new commit of today fixed the battery indicator.
